src/wovnio/wovnphp/core/WovnRequest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+
+namespace Wovnio\Wovnphp\Core;
+
+require_once(__DIR__ . '/WovnLangDirectory.php');
+require_once(__DIR__ . '/WovnLang.php');
+require_once(__DIR__ . '/WovnOption.php');
+require_once(__DIR__ . '/WovnURL.php');
+
+
+class WovnRequest
+{
+    private $scheme; // e.g. HTTP
+    private $host; // e.g. 20.18.166.31, news.network.com, news.com
+    private $path; // e.g. /books/bestseller/1.html, /
+    private $url; // The fully qualified request URL, as-is, e.g. https://www.new.com/breaking/top.html?start=1&wovn=en
+    private $wovnUrl; // The WovnURL object
+    private $query; // The request's query, as an associative array
+    private $lang; // The request's target language code
+
+    private $langDirectory;
+    private $options;
+
+    public function __construct($serverSuperGlobal, $optionConfig, $useStrictMode = true)
+    {
+        $this->options = new WovnOption($optionConfig, $useStrictMode); // Do Not handle exceptions here, let them fail.
+        $this->scheme = $this->parseScheme($serverSuperGlobal);
+        $this->host = $serverSuperGlobal['HTTP_HOST'];
+        $this->path = $this->parseURI($serverSuperGlobal);
+        $this->query = isset($serverSuperGlobal['QUERY_STRING']) ? $serverSuperGlobal['QUERY_STRING'] : '';
+        $this->url = $this->scheme . '://' . $this->host . $this->path . $this->query;
```

Comment:
When I access with `http://testsite.com/?wovn=en`, $this->url will be `http://testsite.com/?wovn=enwovn=en`.

Reply:
Thank you for pointing this out. I didn't realize the REQUEST_URI includes the query string.